client/components/TestResult/index.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'react-bootstrap';
+
+class TestResult extends Component {
+    renderPassingAssertions(assertions) {
+        if (assertions.length === 0) {
+            return (
+                <ul>
+                    <li>No passing assertions.</li>
+                </ul>
+            );
+        }
+        return (
+            <ul>
+                {assertions.map((a, index) => (
+                    <li key={index}>{`${a.pass}: ${a.assertion}`}</li>
+                ))}
+            </ul>
+        );
+    }
+
+    renderFailingAssertions(assertions) {
+        if (assertions.length === 0) {
+            return (
+                <ul>
+                    <li>No failing assertions.</li>
+                </ul>
+            );
+        }
+        return (
+            <ul>
+                {assertions.map((a, index) => (
+                    <li key={index}>{`${a.fail}: ${a.assertion}`}</li>
+                ))}
+            </ul>
+        );
+    }
+
+    renderUnexpected(unexpecteds) {
+        if (unexpecteds.length === 0) {
+            return (
+                <ul>
+                    <li>No unexpected behavior.</li>
+                </ul>
+            );
+        }
+        return (
+            <ul>
+                {unexpecteds.map((u, index) => (
+                    <li key={index}>{u}</li>
+                ))}
+            </ul>
+        );
+    }
```

Comment:
All three of these could be reduced to a single function: 

```js
    renderOutcomeList(outcomes) {
        if (!outcomes.length) {
            outcomes.push('None.');
        }
        return (
            <ul>
                {outcomes.map((outcome, index) => (
                    <li key={useId()}>{outcome}</li>
                ))}
            </ul>
        );
    }
```
(Note that `useId()` assumes that you use the module I recommended above)

Then your code below becomes: 

```js
let passing = command.assertions.reduce((accum, a) => 
    accum.concat(a.pass !== undefined ? [`${a.pass}: ${a.assertion}`] : []), []
);
let failing = command.assertions.reduce((accum, a) => 
    accum.concat(a.fail !== undefined ? [`${a.fail}: ${a.assertion}`] : []), []
);

return (
    <tr key={useId()}>
        <td>{command.command}</td>
        <td>{command.support}</td>
        <td>
            <p>Output: {command.output}</p>
            <p>Passing Assertions:</p>
            {this.renderOutcomeList(
                passing
            )}
            <p>Failing Assertions:</p>{' '}
            {this.renderOutcomeList(
                failing
            )}
            <p>Unexpected Behaviors:</p>{' '}
            {this.renderOutcomeList(
                command.unexpected_behaviors
            )}
        </td>
    </tr>
);
```